internal/stats/apirequest.go

```diff
@@ -22,12 +22,28 @@ type APIRequestMetric struct {
 type APIRequestMetricQuery struct {
 	BaseQuery
 
-	Sender       string `json:"target-sender"`
-	Endpoint     string `json:"target-endpoint"`
-	UserEmail    string `json:"target-user,omitempty"`
-	CourseID     string `json:"target-course,omitempty"`
-	AssignmentID string `json:"target-assignment,omitempty"`
-	Locator      string `json:"target-locator"`
+	AggregationQuery
+
+	IncludeAPIRequestMetricField
+	ExcludeAPIRequestMetricField
+}
+
+type IncludeAPIRequestMetricField struct {
+	Sender       string `json:"include-sender,omitempty"`
+	Endpoint     string `json:"include-endpoint,omitempty"`
+	UserEmail    string `json:"include-user,omitempty"`
+	CourseID     string `json:"include-course,omitempty"`
+	AssignmentID string `json:"include-assignment,omitempty"`
+	Locator      string `json:"include-locator,omitempty"`
+}
+
+type ExcludeAPIRequestMetricField struct {
+	Sender       string `json:"exclude-sender,omitempty"`
+	Endpoint     string `json:"exclude-endpoint,omitempty"`
+	UserEmail    string `json:"exclude-user,omitempty"`
+	CourseID     string `json:"exclude-course,omitempty"`
+	AssignmentID string `json:"exclude-assignment,omitempty"`
+	Locator      string `json:"exclude-locator,omitempty"`
```

Comment:
This seems bananas to me.

Why do we need include/exclude?

And if we do need it, why not just use strings like `GroupByFields`?